File: find_duplicates.rb
#!/usr/bin/ruby
# Written by Forest J. Handford (She/Her)

# This is a Ruby implementation of the ideal solution for a common interview
# question.
class FindDupes
  # Answer for current array (22, 88, 134) are duplicates
  NUMBERS = [1, 88, 18, 88, 22, 134, 95, 134, 5, 8, 88, 22, 59, 134, 3, 42,
             133, 67, 163].freeze
  attr_accessor :count

  def build_count
    @count = {}
    NUMBERS.each do |num|
      if @count[num].nil?
        @count[num] = 1
      else
        @count[num] += 1
      end
    end
  end

  def list_dupes
    build_count
    @count.each do |key, val|
      if val > 1
        puts 'There are ' + val.to_s + ' of number ' + key.to_s +
             ' in the list!'
      end
    end
  end
end

FindDupes.new.list_dupes if $PROGRAM_NAME == __FILE__

File: find_duplicates.py
#!/usr/bin/python
# Written by Forest J. Handford (She/Her)

# This is a Python implementation of the ideal solution for a common interview
# question.
class FindDupes:
    """Class to find duplicates in an ARRAY."""

    @staticmethod
    def build_count():
        count = {}
        numbers = [1, 88, 18, 88, 22, 134, 95, 134, 5, 8, 88, 22, 59, 134, 3,
                   42, 133, 67, 163]
        for num in numbers:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1
        return count

    @staticmethod
    def list_dupes():
        count = FindDupes.build_count()
        for key, val in count.items():
            if val > 1:
                print("There are %i of number %i in the list!" % (val, key))


FindDupes.list_dupes()
